File: 2023/008/part_one.py
import common

if __name__ == "__main__":
    for input_file in common.inputs:
        movement = None
        nodes = {}

        for line in common.read_file(input_file):
            if movement is None:
                movement = line.strip()
            else:
                if not line:
                    continue

                node, left_right = [x.strip() for x in line.split("=")]
                left, right = [x.strip() for x in left_right.split(",")]
                left, right = left[1:], right[:-1]
                nodes[node] = (left, right)

        current_node = "AAA"
        current_movement_index = 0
        number_of_steps = 0

        while current_node != "ZZZ":
            number_of_steps += 1
            next_move = movement[current_movement_index]

            current_movement_index += 1
            if current_movement_index >= len(movement):
                current_movement_index = 0

            if next_move == "L":
                current_node = nodes[current_node][0]
            elif next_move == "R":
                current_node = nodes[current_node][1]
            else:
                raise ValueError("Unknown next move: %s" % (next_move,))

        print(input_file, number_of_steps)

File: kotlin/src/main/kotlin/pt/dmb/adventofcode/y2016/p01/PartOne.kt
package pt.dmb.adventofcode.y2016.p01

import pt.dmb.adventofcode.common.Utilities

class PartOne {
}

fun main() {
    Utilities.readInputFile(2016, 1, scope = "main").forEach { line ->
        val solver = PartOne()
        TODO()
    }
}

/*
import common

NORTH, SOUTH, EAST, WEST = 0, 1, 2, 3
LEFT, RIGHT = "L", "R"


def turn(direction, facing):
    if facing == NORTH:
        return EAST if direction == RIGHT else WEST
    elif facing == SOUTH:
        return WEST if direction == RIGHT else EAST
    elif facing == EAST:
        return SOUTH if direction == RIGHT else NORTH
    elif facing == WEST:
        return NORTH if direction == RIGHT else SOUTH

    raise ValueError("Unknown direction facing %s to turn %s" % (facing, direction))


def walk(location, direction, steps):
    if direction == NORTH:
        return location[0], location[1] - steps
    elif direction == SOUTH:
        return location[0], location[1] + steps
    elif direction == EAST:
        return location[0] + steps, location[1]
    elif direction == WEST:
        return location[0] - steps, location[1]

    raise ValueError("Unknown direction facing %s to walk %s at %s" % (direction, steps, location))


if __name__ == "__main__":
    for input_file in common.inputs:
        print(input_file)

        for line in common.read_file(input_file):
            facing = NORTH
            location = (0, 0)

            for instruction in line.split(","):
                direction, amount = instruction.strip()[0], int(instruction.strip()[1:])
                facing = turn(direction, facing)
                location = walk(location, facing, amount)

            distance = abs(location[0]) + abs(location[1])
            print("Ended at", location, "which is", distance, "away")

Ended at (-151, -147) which is 298 away
 */